src/Type/Definition/InputObjectField.php

```diff
@@ -50,12 +51,9 @@ public function __construct(array $opts)
         $this->config = $opts;
     }
 
-    /**
-     * @return Type&InputType
```

Comment:
should be kept in order not to lower type coverage

Reply:
Unfortunately, it's not that simple. Before this PR, you were able to get away with this annotation, because `$this->type` received its write from an anonymous, [un-typed array iteration](https://github.com/webonyx/graphql-php/blob/master/src/Type/Definition/InputObjectField.php#L47), so it didn't argue with you about the type, and it would take your word for it on the return type, here. Now that I am returning the result of `Schema::resolveType`, the types must agree. If you can suggest a simple fix I'm all ears, but my long-term plan was to re-introduce native asserts in a follow-up PR.

Reply:
I definitely prefer `assert()` rather than discarding type annotation and losing type coverage.

Reply:
Great, I agree, but let me get into that in a follow-up PR and I'll really do it justice all across the board.

Reply:
If you promise to return the types... 😁

Reply:
Scout's honor! 🤞

Reply:
PHP will not allow us to achieve full type-soundness all the way through. That privilege is reserved for other languages like Haskell 😉

In some cases, we should just be pragmatic and accept that a malicious or careless user *might* cause a type annotation to lie, but still offer the precise type hint for the 99% case.

Reply:
> Typed property GraphQL\Type\Schema::$possibleTypeMap must not be accessed before initialization

I appreciate your input, but I think we can do a little better than just jamming the wrong annotations on there--after all, the annotations we have now _are_ correct for the moment--and ignoring the static checker. If you really, really just want to see that `Type&InputType` up there as some form of documentation, then I can do a stan-style cast. But really, I'd rather just save my energy for a proper refactor in a follow-up PR, so let me hear it from @simPod before I do anything else, here...

Reply:
There are a bunch of fine solutions, but i would like to keep at least some reference to what this type should be. If you want a technically correct type, you can still do something like:

```php
/**
 * @return Type that should implement InputType
 */
```

Reply:
You win. I added a temporary cast. 7aba47f

Reply:
Thank you